twilight-gateway/src/shard.rs

```diff
@@ -1164,14 +1165,14 @@ impl Shard {
                 } else {
                     // Can not use `MessageSender` since it is only polled after
                     // the shard is identified.
-                    self.identify_handle = Some(tokio::spawn({
-                        let shard_id = self.id();
-                        let queue = self.config().queue().clone();
-
-                        async move {
-                            queue.request([shard_id.number(), shard_id.total()]).await;
+                    if let Some(handle) = &self.identify_handle {
+                        // if the "old handle" is ready, we must assume it's no longer valid
+                        if handle.is_finished() {
```

Comment:
Can you replace this with a `map_or(false, JoinHandle::is_finished)` call? (possibly preceeded by `as_ref` as determined by the borrow checker)

Reply:
I think `map_or(true, JoinHandle::is_finished)` will do in this context, as we want to create a new join handle if it's None.